simulation/scripts/course_world_model_demo.py

```
@@ -0,0 +1,431 @@
+            )
+        if any(root.iterdir()):
+            raise SystemExit(
+                "refusing to write into a non-empty evidence directory: "
+                f"{root} (evidence artifacts are regenerated whole; use a "
+                "fresh --out directory)"
+            )
+    except OSError as exc:
+        raise SystemExit(
+            f"refusing to write: cannot prove the evidence root is safe: "
+            f"{root} ({exc})"
+        )
```

Ruff (0.16.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)